api/helpers/add-game-to-match.js

```diff
@@ -40,7 +40,8 @@ module.exports = {
       let numPlayer1Wins = 0;
       let numPlayer2Wins = 0;
 
-      for (const priorGame of [ game, ...relevantMatch.games ]) {
+      const priorGames = _.uniqBy([ game, ...relevantMatch.games ], 'id');
```

Comment:
Hit a case where we were re-adding the same game to a match and it was counted twice, causing the match to appear over with only one win. Basic flow was hitting reLogin in the gamestate api as a spectator after the game is over.

This ensures each game only appears once in the list (using id as the thing to compare)